Core CLR Build-Tests definition is ignoring build failures

There's a couple things going on here, @RussKeldorph , @jashook FYI.

- ILAsm commands are (seemingly randomly) failing in the build
- Build-test.cmd does not successfully fail the VSTS Task phase it's in if it throws errors.

Build : Master - 20180705.05 (Product Build)
Failing configurations:
- Windows
  - Build-Tests-R2R-Release
[Mission Control Build Info](https://mc.dot.net/#/product/netcore/master/source/official~2Fcoreclr~2Fmaster~2F/type/build~2Fproduct~2F/build/20180705.05/workItem/Orchestration/analysis/external/Link)

See 'Run build-test.cmd' here: 
https://devdiv.visualstudio.com/DefaultCollection/DevDiv/Default/_build/results?buildId=1832528&view=logs

```
2018-07-05T01:21:23.7362044Z E:\A\_work\48\s\Tools\IL.targets(48,5): error MSB3073: The command "E:\A\_work\48\s\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -EXE  -OUTPUT=E:\A\_work\48\s\bin\tests\obj\Windows_NT.x64.Release\Managed\JIT\Methodical\Boxing\misc\_relnestval_il\_relnestval_il.exe  nestval.il" exited with code 255. [E:\A\_work\48\s\tests\src\JIT\Methodical\Boxing\misc\_relnestval_il.ilproj]
2018-07-05T01:21:23.7362707Z E:\A\_work\48\s\tests\dir.traversal.targets(25,5): error : (No message specified) [E:\A\_work\48\s\tests\src\dirs.proj]
2018-07-05T01:21:23.7363137Z E:\A\_work\48\s\tests\dir.traversal.targets(25,5): error : (No message specified) [E:\A\_work\48\s\tests\build.proj]
2018-07-05T01:21:23.7363420Z     137 Warning(s)
2018-07-05T01:21:23.7363620Z     3 Error(s)
2018-07-05T01:21:23.7363737Z 
2018-07-05T01:21:23.7363909Z Time Elapsed 00:09:28.74
2018-07-05T01:21:24.2575436Z Command execution failed with exit code 1.
2018-07-05T01:21:24.2733117Z BUILDTEST: Error: build failed. Refer to the build log files for details:
2018-07-05T01:21:24.2734626Z     E:\A\_work\48\s\bin\Logs\Tests_Managed_Windows_NT__x64__Release.log
2018-07-05T01:21:24.2735193Z     E:\A\_work\48\s\bin\Logs\Tests_Managed_Windows_NT__x64__Release.wrn
2018-07-05T01:21:24.2735659Z     E:\A\_work\48\s\bin\Logs\Tests_Managed_Windows_NT__x64__Release.err
2018-07-05T01:21:24.2788070Z ##[section]Finishing: Run build-test.cmd
```

This leg fails, resulting in the next one failing as well (since no test archives get produced)

@mattgal I replaced ilasm.exe with a version that just returns 5.  I then issued the command from the log as you suggested.  As expected, `cmd.exe` reports that `build-test.cmd`'s exit code is non-zero.

```
E:\git\coreclr\master>build-test.cmd Release x64 buildagainstpackages runtimeid osx-x64 TargetsNonWindows  Crossgen  -Priority=1 -- /p:OfficialBuildId=20181109-03 /p:OverwriteCoreClrPackageVersion=true /p:IntermediateAzureFeed=https://dotnetbuildoutput.blob.core.windows.net/coreclr-master-20181109-03/DotNet-CoreClr-Master-1832/Release/pkg/index.json

[most output omitted]

tackCommitCommon.cs(17,60): warning CS0618: 'UnmanagedType.Struct' is obsolete: 'Applying UnmanagedType.Struct is unnecessary when marshalling a struct. Support for UnmanagedType.Struct when marshalling a reference type may be unavailable in future releases.' [E:\git\coreclr\master\tests\src\baseservices\threading\commitstackonlyasneeded\DefaultStackCommit.csproj]
managedcom.cs(8,17): warning CS0618: 'ClassInterfaceType.AutoDual' is obsolete: 'Support for IDispatch may be unavailable in future releases.' [E:\git\coreclr\master\tests\src\baseservices\exceptions\regressions\v4.0\706099\ManagedCom.csproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -DLL  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\compilerservices\RuntimeWrappedException\StringThrower\StringThrower.dll  StringThrower.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\compilerservices\RuntimeWrappedException\StringThrower.ilproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -DLL  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\compilerservices\modulector\moduleCctor\moduleCctor.dll  moduleCctor.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\compilerservices\modulector\moduleCctor.ilproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -DLL  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\exceptions\simple\ILHelper\ILHelper.dll  ILHelper.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\exceptions\simple\ILHelper.ilproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -DLL  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\exceptions\simple\VT\VT.dll  VT.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\exceptions\simple\VT.ilproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -EXE  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\exceptions\generics\try-fault-struct01\try-fault-struct01.exe  try-fault-struct01.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\exceptions\generics\try-fault-struct01.ilproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -EXE  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\exceptions\regressions\v1.0\15266\15266.exe  15266.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\exceptions\regressions\v1.0\15266.ilproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -EXE  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\compilerservices\RuntimeFeature\DefaultImplementationsOfInterfaces\DefaultImplementationsOfInterfaces.exe  DefaultImplementationsOfInterfaces.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\compilerservices\RuntimeFeature\DefaultImplementationsOfInterfaces.ilproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -EXE  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\exceptions\generics\try-fault02\try-fault02.exe  try-fault02.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\exceptions\generics\try-fault02.ilproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -EXE  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\exceptions\generics\try-fault03\try-fault03.exe  try-fault03.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\exceptions\generics\try-fault03.ilproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -EXE  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\exceptions\regressions\V1\SEH\asm\FiltFallThru\FiltFallThru.exe  FiltFallThru.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\exceptions\regressions\V1\SEH\asm\FiltFallThru.ilproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -EXE  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\exceptions\generics\try-fault01\try-fault01.exe  try-fault01.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\exceptions\generics\try-fault01.ilproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -EXE  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\exceptions\generics\try-fault-struct03\try-fault-struct03.exe  try-fault-struct03.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\exceptions\generics\try-fault-struct03.ilproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -EXE  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\exceptions\generics\try-fault-struct02\try-fault-struct02.exe  try-fault-struct02.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\exceptions\generics\try-fault-struct02.ilproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -EXE  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\exceptions\generics\try-filter-struct02\try-filter-struct02.exe  try-filter-struct02.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\exceptions\generics\try-filter-struct02.ilproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -EXE  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\exceptions\generics\try-filter-finally01\try-filter-finally01.exe  try-filter-finally01.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\exceptions\generics\try-filter-finally01.ilproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -EXE  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\exceptions\generics\try-filter02\try-filter02.exe  try-filter02.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\exceptions\generics\try-filter02.ilproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -EXE  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\exceptions\simple\fault\fault.exe  fault.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\exceptions\simple\fault.ilproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -EXE  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\exceptions\generics\try-filter-finally02\try-filter-finally02.exe  try-filter-finally02.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\exceptions\generics\try-filter-finally02.ilproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -EXE  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\exceptions\regressions\V1\SEH\asm\Except\Except.exe  Except.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\exceptions\regressions\V1\SEH\asm\Except.ilproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -EXE  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\exceptions\regressions\V1\SEH\asm\FiltCatch\FiltCatch.exe  FiltCatch.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\exceptions\regressions\V1\SEH\asm\FiltCatch.ilproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -EXE  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\exceptions\regressions\V1\SEH\asm\Filter\Filter.exe  Filter.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\exceptions\regressions\V1\SEH\asm\Filter.ilproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -EXE  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\exceptions\regressions\V1\SEH\asm\Finally\Finally.exe  Finally.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\exceptions\regressions\V1\SEH\asm\Finally.ilproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -EXE  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\exceptions\regressions\V1\SEH\asm\NestedFilt\NestedFilt.exe  NestedFilt.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\exceptions\regressions\V1\SEH\asm\NestedFilt.ilproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -EXE  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\exceptions\regressions\V1\SEH\asm\NestedExcept\NestedExcept.exe  NestedExcept.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\exceptions\regressions\V1\SEH\asm\NestedExcept.ilproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -EXE  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\varargs\varargsupport_r\varargsupport_r.exe  varargsupport.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\varargs\varargsupport_r.ilproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG -OPTIMIZE -EXE  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\varargs\varargsupport\varargsupport.exe  varargsupport.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\varargs\varargsupport.ilproj]
E:\git\coreclr\master\Tools\IL.targets(49,5): error MSB3075: The command "E:\git\coreclr\master\Tools\\ilasm\ilasm.exe -QUIET -NOLOGO -DEBUG=OPT -OPTIMIZE -EXE  -OUTPUT=E:\git\coreclr\master\bin\tests\obj\Windows_NT.x64.Release\Managed\baseservices\multidimmarray\rank1array\rank1array.exe  rank1array.il" exited with code 5. Please verify that you have sufficient rights to run this command. [E:\git\coreclr\master\tests\src\baseservices\multidimmarray\rank1array.ilproj]
E:\git\coreclr\master\tests\dir.traversal.targets(25,5): error : (No message specified) [E:\git\coreclr\master\tests\src\dirs.proj]
E:\git\coreclr\master\tests\dir.traversal.targets(25,5): error : (No message specified) [E:\git\coreclr\master\tests\build.proj]
    16 Warning(s)
    29 Error(s)

Time Elapsed 00:01:14.93
BUILDTEST: Error: build failed. Refer to the build log files for details:
    E:\git\coreclr\master\bin\Logs\Tests_Managed_Windows_NT__x64__Release.log
    E:\git\coreclr\master\bin\Logs\Tests_Managed_Windows_NT__x64__Release.wrn
    E:\git\coreclr\master\bin\Logs\Tests_Managed_Windows_NT__x64__Release.err

E:\git\coreclr\master>echo %errorlevel%
1
```

--- a/build-test.cmd
+++ b/build-test.cmd
@@ -343,7 +343,10 @@ for /l %%G in (1, 1, %__BuildLoopCount%) do (
         echo     %__BuildLog%
         echo     %__BuildWrn%
         echo     %__BuildErr%
-        exit /b 1
+        REM This is necessary because of a(n apparent) bug in the FOR /L command.  Under certain circumstances,
+        REM such as when this script is invoke with CMD /C "build-test.cmd", a non-zero exit directly from
+        REM within the loop body will not propagate to the caller.  For some reason, goto works around it.
+        goto     :Exit_Failure
     )
 
     set __SkipPackageRestore=true
@@ -633,3 +636,6 @@ ren "%CORE_ROOT%\temp.ni.dll" %2
     
 echo Successfully precompiled %2
 exit /b 0
+
+:Exit_Failure
+exit /b 1

